Doc/library/sys.rst

```diff
@@ -1300,7 +1300,12 @@ always available.
    Note that as an exception is propagated down the chain of callers, an
    ``'exception'`` event is generated at each level.
 
-   For more information on code and frame objects, refer to :ref:`types`.
+   For more fine-grained usage, it's possible to set a trace function for each
```

Comment:
I think the only thing missing here is some indication that ``frame.f_trace`` is used to store the return value from the global trace function when the frame is entered, and on each trace function invocation after that:
```suggestion
   For more fine-grained usage, it's possible to set a trace function by assigning ``frame.f_trace = tracefunc`` explicitly, rather than relying on it being set indirectly via the return value from an already installed trace function.
```

Without that, folks might be surprised by `frame.f_trace` mysteriously changing out from under them.

Reply:
(Note that that suggestion is intended to replace the whole first sentence)